examples/landau-zener-stuckelberg.ipynb

```diff
@@ -91,58 +91,46 @@
    "source": [
     "# ODE settings (for list-str format)\n",
     "options = Options()\n",
-    "options.rhs_reuse = True\n",
     "options.atol = 1e-6 # reduce accuracy to speed\n",
-    "options.rtol = 1e-5 # up the calculation a bit"
+    "options.rtol = 1e-5 # up the calculation a bit\n",
+    "options.rhs_reuse = True # Compile Hamiltonian only the first time."
    ]
   },
   {
    "cell_type": "code",
    "execution_count": 7,
    "metadata": {},
    "outputs": [],
-   "source": [
-    "# for function-callback style time-dependence\n",
-    "def hamiltonian_t(t, args):\n",
-    "    \"\"\" evaluate the hamiltonian at time t. \"\"\"\n",
-    "    H0 = args[0]\n",
-    "    H1 = args[1]\n",
-    "    w  = args[2]\n",
-    "    return H0 + H1 * np.sin(w * t)"
-   ]
-  },
-  {
-   "cell_type": "code",
-   "execution_count": 8,
-   "metadata": {},
-   "outputs": [],
    "source": [
     "# perform the calculation for each combination of eps and A, store the result\n",
     "# in a matrix\n",
     "def calculate():\n",
     "\n",
     "    p_mat = np.zeros((len(eps_list), len(A_list)))\n",
-    "\n",
-    "    pbar = ProgressBar(len(eps_list))\n",
     "    \n",
+    "    H0 = - delta/2.0 * sx \n",
+    "    \n",
+    "    H1 = - sz/2 \n",
+    "    # Function method\n",
+    "#     H1 = [H1, lambda t, args: args['eps'] ]\n",
+    "    # String method\n",
+    "    H1 = [H1, 'eps']\n",
+    "    \n",
+    "    H2 = sz/2\n",
+    "    # Function method\n",
+    "#     H2 = [H2, lambda t, args: args['A']*np.sin(args['w'] * t)]\n",
+    "    # String method\n",
+    "    H2 = [H2, 'A * sin(w * t)']\n",
+    "    \n",
+    "    H = [H0, H1, H2]\n",
+    "    \n",
+    "    pbar = ProgressBar(len(eps_list))\n",
     "    for m, eps in enumerate(eps_list):\n",
-    "        H0 = - delta/2.0 * sx - eps/2.0 * sz\n",
-    "\n",
     "        pbar.update(m)\n",
-    "\n",
     "        for n, A in enumerate(A_list):\n",
-    "            H1 = (A/2) * sz\n",
-    "\n",
-    "            # function callback format\n",
-    "            #args = (H0, H1, w); H_td = hamiltonian_t\n",
-    "\n",
-    "            # list-str format\n",
-    "            #args = {'w': w}; H_td = [H0, [H1, 'sin(w * t)']]\n",
-    "\n",
-    "            # list-function format\n",
-    "            args = w; H_td = [H0, [H1, lambda t, w: np.sin(w * t)]]\n",
+    "            args = {'w': w, 'A': A, 'eps': eps}\n",
```

Comment:
Thanks for finding and fixing this. How did `args = w` ever work?! O_o

Reply:
I think the last time this notebook was executed was on 2017. I guess a lot has changed in QuTiP since them :upside_down_face:.

Reply:
Indeed. We really need to be able to run these notebooks nightly, but that will have to wait for another day. :)